### ISSUE TITLE
Can I just choose a location for my local storage?

### ISSUE THREAD

**Issue by TrueMegist** [toeverything/AFFiNE#8263]
### Description

I cannot understand how an offline-first app doesn't have an option to choose the location for local storage. Why does AFFiNE think that I need my entire note system and files stored on my main disk, and why should it be in the browser’s local data? Why, in 2002, could I choose where a program would be installed and where its files would go, but in 2024, when installing a PRODUCTIVITY app that promotes SYSTEM AND STRUCTURE, I can’t decide where my data is stored?

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by pengx17** [comment]
In fact this was a feature we used to have, but was removed later. The main reason was that it causeed serious data loss issues when the users put the the storage into sync services like iCloud or Dropbox.
Due to the internal implementation (yjs) we are unable find a proper way to safely merge the synced files with what is stored locally.

**Comment by createchange** [comment]
To be frank, I don't know anything about yjs or Affine, but I did have a bookmark that I led me back to this project. I presently use Obsidian.

Someone built an Obsidian plugin that allows you to point at a git repository for syncing - maybe that is worth exploring? 

https://github.com/Vinzent03/obsidian-git

**Comment by matrix-spec** [comment]
Of course, we all perfectly understand that all editors do this so that the data is stored on their side and users pay a subscription. It's disgusting.

**Comment by Oniryans** [comment]
The fact the option isn't given feels a bit like to force subscription (tho after researches, I may be mistaken). Even if I can't sync real time to work in a group setting (I don't need to, I'm a single user), I should be able to choose where I want my files to be located, be it on another cloud service. I already pay for such things, I don't need nor want to for my personal usage have to get into the pro plan and paying for more services, I don't forcefully have the means. I understand you want and need to make a revenue, but still, as a single user I just want to save my workspaces on my already owned storage, out of NAS (I don't forcefully have the means to either). Is the internal implementation really too much asking in the way it works for it being that bad? Do people really lose a lot of data? Is there really no mean to make it work, any workaround or anything, just file wise? What makes it not possible in fact? (I understood for that it's internal implementation, but I'm no techy person, so I don't understand why the implementation makes it not possible? Something to do with the fact a document can be both text based and a board? The overall way doc works, that ask too much? IDK, I'd appreciate some explanations and to understand) I also saw that obsidian had some unofficial plugins for that kind of stuffs

Edit: Okay, I made research about yjs. Is that a problem still for personal usage? The other clouds can't keep up or something? Can't there be a way for things to be saved without problem when a single user is using it so we can have our own backup?

**Comment by HiroDane** [comment]
Just put up a disclaimer and let single offline users use it. why can that not work?

**Comment by abdulhakim-alshanqiti** [comment]
> Of course, we all perfectly understand that all editors do this so that the data is stored on their side and users pay a subscription. It's disgusting.

i don't know about their intentions but i am literally making an editor and i am struggling with cloud sync

its easy to sync yjs documents if you have a server it gets harder if you don't have one

**Comment by hmanbr** [comment]
Dang, this is the only reason that stops me from using this app. Whiteboard + notes is great, but I want my folder location.

**Comment by ifengzhi** [comment]
For the version i used on linux, the local file may be located at directory `~/.config/AFFiNE`.
Because if I add a large file to the app, this size of this directory will be enlarged.

And in this way i can customize my local storage directory
1. Use “Full backup” in the app to backup my notebook and close the app
2. Delete the `~/.config/AFFiNE` directory.
3. Link the directory you want to   `~/.config/AFFiNE` like: `ln -s /mnt/data/affine/AFFiNE  ~/.config/AFFiNE`
5. Restart the app.
6. Load my notebook in the app.

**Comment by Daniel15** [comment]
>  to force subscription 

@Oniryans  you can self-host the server for free.
